mantidproject/mantid
# Fix No found object warning on Elwin Interface

### Description of work
This PR fixes issue #37770 and other related issues: #37787 and #37788. All three issues are related to the underlying problem of trying to retrieve loaded workspaces onto the elwin interface from the ADS. If for some reason the data on the ADS is modified, the interface tries to retrieve workspaces that don't exist, and depending on the actions on the interface can either crash mantid #37787 or give a warning #37770.  
Additionally, the preview plot is chosen by selecting the workspace to plot and the spectrum index on that workspace from two comboboxes. The spectrum combobox was not retrieving the correct index number, and could not plot spectrum number 0. 

#### Summary of work


I have tried to refactor the Elwin Presenter and Elwin View classes to handle better the communication between view and presenter, and to reduce dependencies with the ADS. The elwin presenter already stores copies of the loaded workspaces in their data model, and it's not necessary to retrieve workspaces from the ADS in order to plot data.
Also, Elwin interface used to have a way to load data through a file finder or through a workspace dialog , but this was removed in #36953 in favor of a multiple file workspace dialog. So there are functions on the presenter and view that are not needed anymore, I have tried to remove these as well.




Fixes #37770, #37787, #37788 




#### Further detail of work


### To test:

 - Check that the test instructions for the issues #37770, #37787, #37788 are not happening anymore. 
 - Follow manual testing instructions for elwin tab: https://developer.mantidproject.org/Testing/Inelastic/DataProcessorTests.html . See there are no issues when running the instructions.
 


*Added release notes*



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
